src/dxgi/dxgi_swapchain.cpp

```diff
@@ -425,8 +425,17 @@ namespace dxvk {
 
     Com<IDXGIOutput1> target;
 
-    if (pTarget)
-      pTarget->QueryInterface(IID_PPV_ARGS(&target));
+    if (pTarget) {
+      DXGI_OUTPUT_DESC desc;
+
+      if (FAILED(pTarget->QueryInterface(IID_PPV_ARGS(&target))))
+        desc.Monitor = m_monitor;
+      else
+        target->GetDesc(&desc);
+
+      if (!m_descFs.Windowed && m_monitor != desc.Monitor && FAILED(this->LeaveFullscreenMode()))
+        Logger::err("DXGI: SetFullscreenState: Failed to leave previous fullscreen mode");
```

Comment:
`LeaveFullscreenMode` already logs errors on its own and we don't do this down below, maybe this can be removed?

Also it's probably better if we got rid of the second `LeaveFullscreenMode` call as well, I know that this currently works but the only reason why it works is because `LeaveFullscreenMode` modifies global state, which makes this more than a little bit confusing.

Reply:
Something like that?

```cpp
    if (pTarget) {
      DXGI_OUTPUT_DESC desc;

      pTarget->QueryInterface(IID_PPV_ARGS(&target));
      target->GetDesc(&desc);

      if (!m_descFs.Windowed && Fullscreen && m_monitor != desc.Monitor) {
        this->LeaveFullscreenMode();
        return this->EnterFullscreenMode(target.ptr());
      }
    }

    if (m_descFs.Windowed && Fullscreen)
      return this->EnterFullscreenMode(target.ptr());
    else if (!m_descFs.Windowed && !Fullscreen)
      return this->LeaveFullscreenMode();
```

Reply:
On second thought, maybe just keep it as-is, just get rid of the error log and return the HRESULT from `LeaveFullscreenMode` properly if that fails for whatever reason.